homeassistant/components/utility_meter/sensor.py

```diff
@@ -413,34 +421,32 @@ def calculate_adjustment(
     def async_reading(self, event: Event):
         """Handle the sensor state changes."""
         old_state: State | None = event.data.get("old_state")
-        new_state: State | None = event.data.get("new_state")
-
-        if new_state and self._state is None:
-            # First state update initializes the utility_meter sensors
-            for sensor in self.hass.data[DATA_UTILITY][self._parent_meter][
-                DATA_TARIFF_SENSORS
-            ]:
-                sensor.start(new_state.attributes.get(ATTR_UNIT_OF_MEASUREMENT))
+        new_state: State = event.data.get("new_state")  # type: ignore[assignment] # a state change event always has a new state
 
         try:
-            if (new_state_state := self._validate_state(new_state)) is None:
+            if (new_state_val := self._validate_state(new_state)) is None:
                 return
         except DecimalException as err:
             if self._sensor_delta_values:
```

Comment:
We can probably get ride of this if and have a single warning message ?

Reply:
Yes agree, I will combine them

Reply:
Yes I agree, will combine them